src/drivers/intel/ssp/ssp.c

```diff
@@ -876,6 +887,12 @@ static void ssp_start(struct dai *dai, int direction)
 
 	if (!(ssp->params.clks_control &
 		 SOF_DAI_INTEL_SSP_CLKCTRL_BCLK_ES)) {
+
+		/* enable TRSE/RSRE before SSE */
+		ssp_update_bits(dai, SSCR1,
+				SSCR1_TSRE | SSCR1_RSRE,
+				SSCR1_TSRE | SSCR1_RSRE);
```

Comment:
don't we need to worry about direction at all?

Reply:
no, this follows the recommendations from our hardware friends.

Note that the existing code has a bug where we already set these bits statically without caring about the direction, see 
https://github.com/thesofproject/linux/issues/2943

we've been doing this for years and nothing bad was detected :-)

Reply:
@plbossart not follow, the link you shared is for SDW port?
It looks wired to me to enable Receive Service request when starting playback stream too.

Reply:
I updated the link above. I will share the information offline.